SQL_Assignments/sql_3.txt

```diff
@@ -0,0 +1,394 @@
+1. Extract the middle 3 characters from the string 'ABCDEFG'.
+
+    select SUBSTRING('ABCDEFG',3,3)
+
+
+
+2. From a table 'Employees' with a column 'FullName', write a query to extract the first name (assuming it's always the first word before a space).
+
+        select SUBSTRING(FullName,1,charindex(' ',fullname+' ')) as firstname from emp 
```

Comment:
Here a space will be there at the end of the output right ?, it should be removed, no.
We can simply write like this no : CHARINDEX(' ', FullName) , why fullname is appended with space ?

Reply:
yes charindex(' ',fullname) is enough.

Reply:
the reason why i wrote fullname+' ' is that if an employee only entered firstname and no lastname then the index of ' ' cannot be found as there is no space between firstname and lastname so before checking the charindex of fullname i just added a space at the end so that if there is only one word then it will be displayed as such